examples/proxy/index.js

```diff
@@ -0,0 +1,47 @@
+const { Pool, Client } = require('../../')
+const http = require('http')
+const proxy = require('./proxy')
+
+const client = new Pool('http://localhost:4001', {
+  connections: 256,
+  pipelining: 1
+})
+
+Promise.all([
+  new Promise(resolve => {
+    // Proxy
+    http.createServer(async (req, res) => {
+      try {
+        await proxy({ req, res, proxyName: 'example' }, client)
+      } catch (err) {
+        if (res.headersSent) {
+          res.destroy(err)
+        } else {
+          for (const name of res.getHeaderNames()) {
+            res.removeHeader(name)
+          }
+          res.writeHead(err.statusCode || 500)
+          res.end()
+        }
+      }
+    }).listen(4000, resolve)
+  }),
+  new Promise(resolve => {
+    // Upstream
+    http.createServer((req, res) => {
+      res.end('hello world')
+    }).listen(4001, resolve)
+  })
+]).then(async () => {
+  const client = new Client('http://localhost:4000')
+  const { body } = await client.request({
+    method: 'GET',
+    path: '/'
+  })
+
+  for await (const chunk of body) {
+    console.log(String(chunk))
+  }
+})
```

Comment:
I would recommend to just avoid `.then` in this file and use a `async run() {}` function at the top.